tianshou/policy/modelbase/psrl.py

```diff
@@ -0,0 +1,198 @@
+    """Implementation of Posterior Sampling Reinforcement Learning Model.
+
+    :param np.ndarray p_prior: dirichlet prior (alphas),
+        shape: (n_action, n_state, n_state).
+    :param np.ndarray rew_mean_prior: means of the normal priors of rewards,
+        shape: (n_state, n_action)
+    :param np.ndarray rew_std_prior: standard deviations of the normal
+        priors of rewards, shape: (n_state, n_action).
+
+    .. seealso::
+
+        Please refer to :class:`~tianshou.policy.BasePolicy` for more detailed
+        explanation.
+    """
+
+    def __init__(
+        self,
+        p_prior: np.ndarray,
+        rew_mean_prior: np.ndarray,
+        rew_std_prior: np.ndarray,
+    ) -> None:
+        self.p = p_prior
+        self.n_action = len(self.p)
+        self.n_state = len(self.p[0])
+        self.rew_mean = rew_mean_prior
+        self.rew_std = rew_std_prior
+        self.rew_count = np.zeros_like(rew_mean_prior)
+        self.p_ml = None
+        self.sample_rew = None
+        self.policy = None
+        self.updated = False
+
+    def observe(
+        self, p: np.ndarray, rew_sum: np.ndarray, rew_count: np.ndarray
+    ) -> None:
+        """Add data into memory pool.
+        :param np.ndarray p: the number of observations,
+            shape: (n_action, n_state, n_state).
+        :param np.ndarray rew_sum: total rewards,
+            shape: (n_state, n_action)
+        :param np.ndarray rew_count: the number of rewards,
+            shape: (n_state, n_action).
+        """
+        self.updated = False
+        self.p += p
+        sum_count_nonzero = np.where(self.rew_count + rew_count == 0,
+                                     1, self.rew_count + rew_count)
+        rew_count_nonzero = np.where(rew_count == 0, 1, rew_count)
+        self.rew_mean = np.where(self.rew_count == 0,
+                                 np.where(rew_count == 0, self.rew_mean,
+                                          rew_sum / rew_count_nonzero),
+                                 (self.rew_mean * self.rew_count + rew_sum)
+                                 / sum_count_nonzero)
+        self.rew_std *= np.where(self.rew_count == 0, 1,
+                                 self.rew_count) / sum_count_nonzero
+        self.rew_count += rew_count
+
+    def get_p_ml(self) -> np.ndarray:
+        return self.p / np.sum(self.p, axis=-1, keepdims=True)
+
+    def sample_from_rew(self) -> np.ndarray:
+        sample_rew = np.random.randn(*self.rew_mean.shape)
+        sample_rew = sample_rew * self.rew_std + self.rew_mean
+        return sample_rew
+
+    def solve_policy(self) -> np.ndarray:
+        self.updated = True
+        self.p_ml = self.get_p_ml()
+        self.sample_rew = self.sample_from_rew()
+        self.policy = self.value_iteration(self.p_ml, self.sample_rew)
+        return self.policy
+
+    @staticmethod
+    def value_iteration(p: np.ndarray, rew: np.ndarray,
+                        epsilon: float = 0.01) -> np.ndarray:
```

Comment:
Could you please add some docstring here? For example, what does `p` stands for? (I know it is trans prob)